remove literal PW (#5016)
dotnet/EntityFramework.Docs: entity-framework/core/cli/dotnet.md

The following example skips scaffolding an `OnConfiguring` method. This can be useful when you want to configure the DbContext outside of the class. For example, ASP.NET Core apps typically configure it in Startup.ConfigureServices.

```dotnetcli

> dotnet ef dbcontext scaffold "Server=(localdb)\mssqllocaldb;Database=Blogging;User Id=myUsername;Password=--;" Microsoft.EntityFrameworkCore.SqlServer --no-onconfiguring

Author: Is that still a legal pw?

Reviewer: Uh, I'm not sure - but it's the same password you had before: you had `Password=--` and then an additional `;pw` which wasn't part of the password, since `;` is the connection string delimiter. And since `pw` isn't a valid option, the connection string was invalid.

Author: > Uh, I'm not sure - but it's the same password you had before: you had `Password=--` and then an additional `;pw` which wasn't part of the password, since `;` is the connection string delimiter. And since `pw` isn't a valid option, the connection string was invalid.

that was my intention. 

I should have added:

In the preceding code, replace `--;pw;` with a valid password.

Reviewer 2: If *** is a valid password you can't use it. You have to use something that can never be valid, even better if it throws exceptions

Author: > If *** is a valid password you can't use it. You have to use something that can never be valid, even better if it throws exceptions

Sorry @blowdart, I see you resolved the issue.

> dotnet ef dbcontext scaffold "Server=(localdb)\mssqllocaldb;Database=Blogging;User Id=myUsername;Password=\0;" Microsoft.EntityFrameworkCore.SqlServer --no-onconfiguring

Author: ```suggestion
dotnet ef dbcontext scaffold "Server=(localdb)\mssqllocaldb;Database=Blogging;User Id=myUsername;Password=\0;" Microsoft.EntityFrameworkCore.SqlServer --no-onconfiguring
```

In the preceding code, replace `\0` with a valid strong password.

> dotnet ef dbcontext scaffold "Server=(localdb)\mssqllocaldb;Database=Blogging;User Id=myUsername;Password=--;pw;" Microsoft.EntityFrameworkCore.SqlServer --no-onconfiguring

Reviewer: @Rick-Anderson this makes the connection string invalid (because of the `;pw`). What's the exact intent here, is it to change `myPassword` to `--` or similar?

The following example skips scaffolding an `OnConfiguring` method. This can be useful when you want to configure the DbContext outside of the class. For example, ASP.NET Core apps typically configure it in Startup.ConfigureServices.

```dotnetcli

> dotnet ef dbcontext scaffold "Server=(localdb)\mssqllocaldb;Database=Blogging;User Id=myUsername;Password=\0;" Microsoft.EntityFrameworkCore.SqlServer --no-onconfiguring

Author: ```suggestion
dotnet ef dbcontext scaffold "Server=(localdb)\mssqllocaldb;Database=Blogging;User Id=myUsername;Password=***" Microsoft.EntityFrameworkCore.SqlServer --no-onconfiguring
```

> In the preceding code, replace `\0` with a valid strong password.

Author: That's probably still a valid password. How about I use `;|;` or `;A;` and tell them to replace `;A` with a ...

Reviewer 3: Why not simply `***`?

Author: > Why not simply `***`?

Because that's a valid PW and can be copy/pasted. And customer copy/paste.

> In the preceding code, replace `\0` with a valid strong password.

Author: ```suggestion
In the preceding code, replace `***` with a valid strong password, followed by a semicolon `;`, for example, `password=<A strong password>;`
```

> In the preceding code, replace `***` with a valid strong password, followed by a semicolon `;`, for example, `password=<A strong password>;`

Author: At least this makes them do a copy/paste/copy `password=<A strong password>;/edit

`***;` by itself is a valid password and can be copy/pasted - right? If not, I'll revert to `***;`

Reviewer: @Rick-Anderson yes, `***` can be a valid password. So can the string `\0`.

Having something like `...;User Id=myUsername;Password=;A;` doesn't mean that there's an invalid password `;A;`.  Since semicolon is the separator for connection string, this means there's an empty password (which in theory might also be valid? I'm not sure), and then an invalid connection string option `A` (which will make the connection fail).

I'm not sure what exactly is the idea here... Yes, documentation placeholders for password are generally legal passwords - that seems quite standard; what exactly is the problem with using `***` as a placeholder?

Author: > @Rick-Anderson yes, `***` can be a valid password. So can the string `\0`.
> 
> Having something like `...;User Id=myUsername;Password=;A;` doesn't mean that there's an invalid password `;A;`. Since semicolon is the separator for connection string, this means there's an empty password (which in theory might also be valid? I'm not sure), and then an invalid connection string option `A` (which will make the connection fail).

Perfect, that's what we want. Copy/paste fail.

> I'm not sure what exactly is the idea here... Yes, documentation placeholders for password are generally legal passwords - that seems quite standard; what exactly is the problem with using `***` as a placeholder?

Right, and smoking was standard 50 years ago, even among MD. Lots of bad security practices were use in the past. The  SFI mantra is remove insecure practices.

Reviewer: > Perfect, that's what we want. Copy/paste fail.

I disagree. You're saying our doc samples should show a connection string that's invalid, in a way that has nothing to do with the password, by adding an arbitrary invalid `A` option **just** to make the connection string fail. That just seems wrong.

> Right, and smoking was standard 50 years ago, even among MD. Lots of bad security practices were use in the past. The SFI mantra is remove insecure practices.

What exact problem are you trying to solve? How exactly is having `***` as a password placeholder in our docs insecure? Is there some guideline you can point to that says we need to do this?

> dotnet ef dbcontext scaffold "Server=(localdb)\mssqllocaldb;Database=Blogging;User Id=myUsername;Password=***" Microsoft.EntityFrameworkCore.SqlServer --no-onconfiguring
> 
> In the preceding code, replace `***` with a valid strong password, followed by a semicolon `;`, for example, `password=<A strong password>;`

Author: ```suggestion
dotnet ef dbcontext scaffold "Server=(localdb)\mssqllocaldb;Database=Blogging;User Id=myUsername;Password=;;;" Microsoft.EntityFrameworkCore.SqlServer --no-onconfiguring

In the preceding code, replace `Password=;;;` with a valid strong password, followed by a semicolon `;`, for example, `password=<A strong password>;`
```

> dotnet ef dbcontext scaffold "Server=(localdb)\mssqllocaldb;Database=Blogging;User Id=myUsername;Password=;;;" Microsoft.EntityFrameworkCore.SqlServer --no-onconfiguring
> 
> In the preceding code, replace `Password=;;;` with a valid strong password, followed by a semicolon `;`, for example, `password=<A strong password>;`

Author: ```suggestion
dotnet ef dbcontext scaffold "Server=(localdb)\mssqllocaldb;Database=Blogging;Integrated Security=true;" Microsoft.EntityFrameworkCore.SqlServer --no-onconfiguring
```
